app/models/polymorphic/comment.rb

```diff
@@ -53,7 +53,8 @@ def subscribe_user_to_entity(u = user)
   # Notify subscribed users when a comment is added, unless user created this comment
   def notify_subscribers
     commentable.subscribed_users.reject { |user_id| user_id == user.id }.each do |subscriber_id|
-      if subscriber = User.find_by_id(subscriber_id)
+      subscriber = User.find_by_id(subscriber_id)
```

Comment:
Assuming this is an activerecord relation; not an array of ids (hard to tell), the below
```
commentable.subscribed_users.reject { |user_id| user_id == user.id }
```
should be refactored to a scope or `.where.not(user+id: user.id)`; so that you don't do a loop of user objects you already have.

If its an array of user ids; it should still be refactored to something like
```
other_users = User.where(id: commentable.subscribed_users.reject { |user_id| user_id == user.id }.collect(&:subscriber_id))
```
Then you can further add scopes for currently_emailable applying your conditions below.

Avoid N+1 issues